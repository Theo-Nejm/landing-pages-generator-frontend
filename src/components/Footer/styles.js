import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    p,
    a,
    i,
    b {
      font-size: ${theme.fonts.sizes.xxsmall};
      text-align: center;
    }
  `}
`;
