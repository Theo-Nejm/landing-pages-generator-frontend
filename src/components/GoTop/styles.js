import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    color: #fff;
    background: ${theme.colors.primaryColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    z-index: 4;
    opacity: 0.85;
  `}
`;
