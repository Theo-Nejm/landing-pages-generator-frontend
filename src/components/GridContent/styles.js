import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
  `}
`;

export const HTMLContainer = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xlarge} 0;
  `}
`;
