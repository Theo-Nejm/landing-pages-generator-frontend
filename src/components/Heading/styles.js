import styled, { css } from 'styled-components';

const titleSize = {
  xsmall: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xsmall};
    `,
  small: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.small};
    `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  large: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.large};
    `,
  xlarge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xlarge};
    `,
  xxlarge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xxlarge};
    `,
  huge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.huge};
      ${mediaFont(theme)};
    `,
  xhuge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xhuge};
      ${mediaFont(theme)};
    `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.fonts.sizes.large};
  }
`;

const titleCase = (upperCase) => css`
  text-transform: ${upperCase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, upperCase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : '#FFF'};
    ${titleSize[size](theme)};
    ${titleCase(upperCase)};
  `}
`;
