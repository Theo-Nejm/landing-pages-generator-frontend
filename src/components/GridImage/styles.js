import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    h2 {
      font-size: 4.2rem;
    }

    ${TextComponent} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const Element = styled.div`
  ${() => css`
    overflow: hidden;

    > img {
      width: 100%;
      transition: all 0.3s ease-in-out;

      &:hover {
        transform: scale(1.2) rotate(10deg);
      }
    }
  `}
`;
