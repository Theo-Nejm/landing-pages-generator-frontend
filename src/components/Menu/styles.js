import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, isOpened }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    width: 100vw;
    transition: all .2s ease-in-out;
    height: 7rem;
    border-bottom: 1px solid ${theme.colors.mediumGray};

    > ${SectionContainer} {
      padding-top: 0;
      padding-botom: 0;
    }

    @media ${theme.media.medium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;

      ${isOpened && menuVisible}
    }
  }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 7rem;

    @media ${theme.media.medium} {
      display: flex;
      flex-flow: column nowrap;
      text-align: center;
      align-items: center;
      justify-content: center;
      height: 100vh;

      img,
      h1 {
        margin-bottom: 2rem;
      }
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 3.5rem;
    height: 3.5rem;
    cursor: pointer;
    background: none;
    border: none;
    color: ${theme.colors.primaryColor};
    display: none;
    @media ${theme.media.medium} {
      display: flex;
    }
  `}
`;
