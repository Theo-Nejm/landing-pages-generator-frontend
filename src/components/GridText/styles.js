import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column nowrap;
    margin-top: 6rem;

    h2 {
      font-size: 4.2rem;
    }

    ${TextComponent} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const Element = styled.div`
  ${({ theme }) => css`
    ${Heading} {
      position: relative;
      left: 5rem;
    }

    ${Heading}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xhuge};
      left: -5rem;
      top: -2rem;
      transform: rotate(7.5deg);
    }
  `}
`;
