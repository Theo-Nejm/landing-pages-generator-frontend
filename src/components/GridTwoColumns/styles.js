import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
  background: none;
  @media ${theme.media.medium} {
    margin-bottom: ${theme.spacings.medium};
  }
    }
  `}
`;

export const ImageContainer = styled.div`
  > img {
    width: 100%;
  }
`;
