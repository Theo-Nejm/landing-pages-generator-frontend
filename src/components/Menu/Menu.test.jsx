import { fireEvent, screen } from '@testing-library/dom';
import { Menu } from '.';
import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';
import linksMock from '../NavBar/mock';

const logoData = {
  text: 'Logo',
  link: '#target',
};
describe('<Menu />', () => {
  it('should render <Menu />', () => {
    renderTheme(<Menu links={linksMock} logoData={logoData} />);
    expect(
      screen.getByRole('navigation', { name: 'Main menu' }),
    ).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <Menu links={linksMock} logoData={logoData} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render mobile menu and o/c buttons', () => {
    renderTheme(<Menu links={linksMock} logoData={logoData} />);
    const button = screen.getByLabelText('Open/Close menu');
    const menuContainer = button.nextSibling;

    expect(button).toHaveStyleRule('display', 'none');

    expect(button).toHaveStyleRule('display', 'flex', {
      media: theme.media.medium,
    });

    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.medium,
    });

    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should render o/c buttons properly', () => {
    renderTheme(<Menu links={linksMock} logoData={logoData} />);
    const button = screen.getByLabelText('Open/Close menu');

    fireEvent.click(button);
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();
    expect(screen.queryByLabelText('Open menu')).not.toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
    expect(screen.queryByLabelText('Close menu')).not.toBeInTheDocument();
  });

  it('should close menu when click on container', () => {
    renderTheme(<Menu links={linksMock} logoData={logoData} />);
    const button = screen.getByLabelText('Open/Close menu');
    const menuContainer = button.nextSibling;

    fireEvent.click(button);
    fireEvent.click(menuContainer);
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.medium,
    });
  });

  it('should not render links when no links were passed as props', () => {
    const { container } = renderTheme(<Menu logoData={logoData} />);
    expect(
      screen.queryByRole('navigation', { name: 'Main menu' }).firstChild,
    ).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
