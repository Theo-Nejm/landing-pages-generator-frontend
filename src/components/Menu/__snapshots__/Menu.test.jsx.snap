// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match snapshot 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  width: 100vw;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  height: 7rem;
  border-bottom: 1px solid #ddd;
}

.c2 > .c3 {
  padding-top: 0;
  padding-botom: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 7rem;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 3.5rem;
  height: 3.5rem;
  cursor: pointer;
  background: none;
  border: none;
  color: #0a1128;
  display: none;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  height: 2.4rem;
}

.c6 {
  color: #0a1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c9 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c9::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 25%;
  height: 1px;
  width: 50%;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c9:hover::after {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

@media screen and (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }
}

@media screen and (max-width:768px) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    text-align: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    height: 100vh;
  }

  .c5 img,
  .c5 h1 {
    margin-bottom: 2rem;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (max-width:768px) {
  .c8 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6"
        >
          <a
            class="c7"
            href="#target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c8"
        >
          <a
            class="c9"
            href="#target-1"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="c9"
            href="#target-2"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="c9"
            href="#target-3"
            target="_self"
          >
            Link 3
          </a>
          <a
            class="c9"
            href="#target-4"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="c9"
            href="#target-5"
            target="_self"
          >
            Link 5
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Menu /> should not render links when no links were passed as props 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  width: 100vw;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  height: 7rem;
  border-bottom: 1px solid #ddd;
}

.c2 > .c3 {
  padding-top: 0;
  padding-botom: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 7rem;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 3.5rem;
  height: 3.5rem;
  cursor: pointer;
  background: none;
  border: none;
  color: #0a1128;
  display: none;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  height: 2.4rem;
}

.c6 {
  color: #0a1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

@media screen and (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }
}

@media screen and (max-width:768px) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    text-align: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    height: 100vh;
  }

  .c5 img,
  .c5 h1 {
    margin-bottom: 2rem;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (max-width:768px) {
  .c8 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6"
        >
          <a
            class="c7"
            href="#target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c8"
        />
      </div>
    </div>
  </div>
</div>
`;
